### ISSUE TITLE
Add 8th volume fails

### ISSUE THREAD

**Issue by scambra** [fog/fog-vsphere#146]
Exception is raised when filling id and other attributes after creating in vmware, but volume is actually created.

Here is a extract of save method:

```ruby
        def save
          raise Fog::Errors::Error.new('Resaving an existing object may create a duplicate') if persisted?
          requires :server_id, :size, :datastore

          set_unit_number

          data = service.add_vm_volume(self)

          if data['task_state'] == 'success'
            if self.unit_number >= 7
              self.unit_number += 1
            end

            # We have to query vSphere to get the volume attributes since the task handle doesn't include that info.
            created = server.volumes.all.find { |volume| volume.unit_number == self.unit_number }
```

Save method calls set_unit_number, which calls calculate_free_unit_number. That method avoids setting unit_number to 7, but after volume is created, unit_number is increased by 1 and created volume is not found.

Is it this code needed? Because I think it's breaking adding 8th volume:

```ruby
            if self.unit_number >= 7
              self.unit_number += 1
            end
```

**Comment by scambra** [comment]
I know that, but calculate_free_number is taken 7 out, so after creating a volume with unit_number 6, it will assign unit_number 8. Then, after volume is created, unit_number is increased to 9, so created volume is not found.

I don't know if increasing unit_number is needed in case user assigns unit_number manually, instead of relying on calculate_free_number

**Comment by timogoebel** [comment]
@scambra: I belive this is a bug and it's not required to update the unit_number when it is assigned manually. Do you mind opening a PR? :-)

**Comment by scambra** [comment]
fixed in #163
